metaflow/plugins/kfp/kfp.py

```diff
@@ -66,54 +68,48 @@
 
 @dataclass
 class MetaflowBootstrapVars:
-    clean_volume_cmd: str
     environment_type: str
     flow_name: str
     logger_type: str
     monitor_type: str
     namespace: str
-    need_split_index: bool
-    step_name: str
     tags: List[str]
-    task_id: str
-    task_id_template: str
-    user_code_retries: int
+    compile_time_bootstrap_cmd: str
+    run_time_bootstrap_cmd: str
 
 
 class KfpComponent(object):
     def __init__(
         self,
-        name: str,
-        metaflow_bootstrap_vars: MetaflowBootstrapVars,
+        step_name: str,
+        clean_volume_cmd: str,
+        need_split_index: str,
+        task_id: str,
+        task_id_template: str,
+        user_code_retries: str,
         total_retries: int,
         resource_requirements: Dict[str, str],
         kfp_decorator: KfpInternalDecorator,
         accelerator_decorator: AcceleratorDecorator,
         environment_decorator: EnvironmentDecorator,
     ):
-        self.name = name
-        self.clean_volume_cmd = metaflow_bootstrap_vars.clean_volume_cmd
-        self.task_id = metaflow_bootstrap_vars.task_id
-        self.task_id_template = metaflow_bootstrap_vars.task_id_template
-        self.step_name = metaflow_bootstrap_vars.step_name
-        self.flow_name = metaflow_bootstrap_vars.flow_name
+        self.step_name = step_name
+        self.clean_volume_cmd = clean_volume_cmd
+        self.need_split_index = need_split_index
+        self.task_id = task_id
+        self.task_id_template = task_id_template
+        self.user_code_retries = user_code_retries
```

Comment:
This is cleaner, how ever these new variables are still Metaflow bootstrap variables, and not necessarily at the level of a `KfpComponent`.

This is why I suggested they be encapsulated in

```
self.metaflow_bootstrap_vars = metaflow_bootstrap_vars
```

Reply:
see https://github.com/zillow/metaflow/pull/137/files#r748602003